src/classes/PMT_Payment_TEST.cls

```diff
@@ -257,12 +257,118 @@ private class PMT_Payment_TEST {
         System.assertEquals(closedWonStage, opp.StageName);
     }
 
+
+    /*******************************************************************************************************
+     * @description     Validate that payment records are not both paid and written off
+     *******************************************************************************************************/
+    @isTest
+    private static void verifyPaymentsAreNotBothPaidAndWrittenOff() {
```

Comment:
Can you split this up into separate tests so that each test is for only one scenario? It looks like this tests a valid update, an invalid update, a valid insert, and then an invalid insert

Reply:
Thanks @seanpat09, I've split the test up and fixed the case on those asserts.